Improve Restate Server deployment docs (#60)
restatedev/docs-restate: docs/cloud/connecting-services.mdx

Reviewer: not part of your changes, but spotted a missing slash here:
```suggestion
  -p 9070:9070 \
```

View the full spec of the `RestateCloudEnvironment` CRD as [pkl file](https://github.com/restatedev/restate-operator/blob/main/crd/RestateCloudEnvironment.pkl) or less-readable [yaml file](https://github.com/restatedev/restate-operator/blob/main/crd/restatecloudenvironments.yaml).

<Info>
    [Learn more about how the tunnel works below.](/cloud/connecting-services#how-does-the-tunnel-work)
</Info>

### Create your RestateDeployment

Create a `RestateDeployment` manifest for your service, as described in the [deployment documentation](/services/deploy/kubernetes#deployment-with-restate-operator), and specify the cloud environment to register with in the `spec`:
```yaml
spec:
  restate:
    register:
      cloud: my-cloud-environment
```

Once you apply the manifest, the Restate Operator will set up the secure tunnel and automatically register your service with Restate Cloud.

## Serverless Functions and other public endpoints

Restate can invoke your services over the public internet.
This is useful if you are deploying your services as serverless functions, e.g. on Vercel, Cloudflare Workers, or Deno Deploy.

> Or if you have services behind public endpoints that you want to connect to Restate Cloud.

Reviewer: ```suggestion
You can also securely deploy Restate services you operate as directly reachable endpoints.
```

<Info>
    For AWS Lambda, please see the [AWS Lambda functions](#aws-lambda-functions) section below.
</Info>

> It is important to secure access to your publicly accessible services so that only Restate can call them.

Reviewer: Maybe this sounds less scary?
```suggestion
It is important to secure access to your publicly routable services so that only the designated Restate environment can call them.
```

### Setting up the invoker role

Restate Cloud must assume an AWS identity with the permission to perform `lambda:InvokeFunction`, within the same AWS account where the Lambda function is deployed.

Note that the Restate Cloud role is separate from your Lambda function’s execution role.
The execution role is used by your function to perform its tasks, while the invoker role grants Restate Cloud permission to invoke your service handler functions — and nothing else.

You can set up the role either by creating an IAM role yourself or by using the Restate CDK construct library.

#### IAM trust policy for Restate Cloud
To create a new IAM role, you can copy the trust policy directly from the Restate Cloud UI, by visiting [Developers > Security > AWS Lambda](https://cloud.restate.dev/to/developers/integration#lambda).
This trust policy allows the Restate Cloud principal (e.g. `us.restate.cloud`) to assume the role, but only on behalf of the specified environment ID.

#### Using AWS CDK

When you use the [Restate CDK construct library](https://github.com/restatedev/cdk) to deploy
Lambda handlers, the provided invoker role will automatically be granted access

> to invoke the corresponding functions.

Reviewer: Maybe the npmjs page is a more appropriate place to direct people? Your call, both places show the README quite prominently.

```suggestion
You can use AWS CDK together with the [Restate CDK construct library](https://www.npmjs.com/package/@restatedev/restate-cdk) to deploy
Lambda handlers, manage the invoker role, and automatically register Lambda-backed services with a Restate (Cloud) environment.
```

  runtime: lambda.Runtime.NODEJS_LATEST,
  code: lambda.Code.fromAsset("../handlers/dist/"),
  handler: "bundle.handler",
});

const environment = new RestateCloudEnvironment(stack, "CloudEnv", {
  environmentId: process.env.RESTATE_ENV_ID! as EnvironmentId,
  apiKey: new secrets.Secret(stack, "RestateCloudApiKey", {
    secretStringValue: cdk.SecretValue.unsafePlainText(process.env.RESTATE_API_KEY!),
  }),
});

const deployer = new ServiceDeployer(stack, "ServiceDeployer", {
  code: lambda.Code.fromAsset(path.join(__dirname, "../../../dist/register-service-handler")),
  removalPolicy: cdk.RemovalPolicy.DESTROY,
});

deployer.deployService("Greeter", handler.currentVersion, environment);

new cdk.CfnOutput(stack, "RestateIngressUrl", { value: environment.ingressUrl });

app.synth();
```
<GitHubLink url={"https://github.com/restatedev/cdk/blob/main/test/e2e/stacks/restate-cloud-lambda-stack.ts"}/>

- The `RESTATE_ENV_ID` can be found in the Restate Cloud UI in the top left corner when you select your environment.
This ID starts with `env_`.

- For the `RESTATE_API_KEY`, create an API key via `Developers` tab of the Restate Cloud UI and give it `Full` permissions.

> This trust policy allows the Restate Cloud principal (e.g. `us.restate.cloud`) to assume the role, but only on behalf of the specified environment.

Reviewer: ```suggestion
The `RestateCloudEnvironment` automatically creates an IAM role with an appropriate trust policy for the matching Restate Cloud environment. The `ServiceDeployer` construct automatically grants this role permission to invoke any AWS Lambda functions registered with the environment. For more details, consult the relevant [Restate CDK](https://www.npmjs.com/package/@restatedev/restate-cdk) constructs' API documentation.
```

### Register your Lambda function

Once you have created the invoker role, you can register your Lambda through the new role:

```shell
restate deployments register <LAMBDA_FUNCTION_ARN> --assume-role-arn <ROLE_ARN>
```

If something isn't working, the environment logs in the cloud UI may help you find the issue.

<Info>
    **Securing your services:**

    If your Lambda has an appropriate trust policy as described above, you do not
    need to secure incoming requests any further. If you choose to however, the
    identity verification checks will work on Lambda endpoints as well.
</Info>

## Services in private environments

You can connect services that run in a locked-down private environment to Restate Cloud using a secure tunnel.
This allows you to expose your services to Restate Cloud without opening up ports to the public internet.
The tunnel will also act as an authenticating proxy to Restate Cloud and make it appear as if your Restate Cloud environment is inside your private network.

> This allows you to use native access control mechanisms, like VPC Security Groups and Kubernetes network policies, to manage access to your environment.

Reviewer: ```suggestion
This allows you to use native access control mechanisms, like VPC Security Groups and Kubernetes network policies, to manage access to your Cloud environment - as well as from the environment to the rest of your network by restricting the internal services that the tunnel client can connect to.
```

Author: ```suggestion
This allows you to use native access control mechanisms, like VPC Security Groups and Kubernetes network policies, to manage access to your Cloud environment. It also lets you control connections from the Cloud environment to the rest of your network by restricting which internal services the tunnel client can connect to.
```

@pcholakov Maybe 2 sentences?
